packages/toggle/src/Toggle.tsx

```diff
@@ -24,431 +25,416 @@ export const Mode = {
-      color: white;
-    }
-  `,
-);
-
-const offLabelStyle = cx(
-  baseLabelStyle,
-  css`
-    right: 6px;
-    color: ${uiColors.gray.base};
-  `,
-);
-
-const getStatefulStyles = (states: StateForStyle) => ({
-  slider: getSliderStyles(states),
-  groove: getGrooveStyles(states),
-  container: getContainerStyles(states),
-});
+    onLabel: css`
+      color: ${uiColors.white};
+      text-shadow: 0 0 2px ${uiColors.blue.base};
+    `,
+  },
+} as const;
 
 interface BaseToggleProps {
+  /**
+   * Sets the size of the toggle.
+   *
+   * default: `'default'`
+   */
   size?: Size;
+
+  /**
+   * Determines if the Toggle will render the dark mode styles.
+   *
+   * default: `false`
+   */
   darkMode?: boolean;
-}
 
-type ToggleProps = BaseToggleProps &
-  Omit<HTMLElementProps<'input', never>, keyof BaseToggleProps>;
+  /**
+   * Sets the checked state of the Toggle.
+   */
+  checked?: boolean;
+
+  /**
+   * Disables the Toggle.
+   *
+   * default: `false`
+   */
+  disabled?: boolean;
+
+  /**
+   * `onChange` fires when the internally-managed `checked` state of the component is updated. Receives the updated checked state of the toggle as its first argument, and the associated mouse event as the second.
+   */
+  onChange?: (
+    checked: boolean,
+    mouseEvent: React.MouseEvent<HTMLButtonElement>,
+  ) => void;
+
+  /**
+   * The event handler function for the 'onchange' event. Receives the associated `event` object as the first argument.
+   */
+  onClick?: React.MouseEventHandler<HTMLButtonElement>;
+
+  /**
+   * Adds a className to the outermost element.
+   */
+  className?: string;
+}
 
-const idAllocator = IdAllocator.create('toggle');
+type ToggleProps = Either<
+  BaseToggleProps &
+    Omit<HTMLElementProps<'button', never>, keyof BaseToggleProps>,
+  'aria-label' | 'aria-labelledby'
+>;
 
 function Toggle({
-  name,
   className,
   size = Size.Default,
   darkMode = false,
   disabled = false,
-  onChange: onChangeProp = () => {},
+  onChange: onChangeProp,
+  onClick: onClickProp,
   checked: checkedProp,
-  id: idProp,
   ...rest
 }: ToggleProps) {
-  const [checked, setChecked] = useState(false);
-  const toggleId = useMemo(() => idProp ?? idAllocator.generate(), [idProp]);
-  const normalizedChecked = checkedProp || checked;
+  validateAriaLabelling(rest, 'Toggle');
 
-  const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
-    if (onChangeProp) {
-      onChangeProp(e);
-    }
+  const [buttonElement, setButtonElement] = useState<HTMLElement | null>(null);
+  const [checked, setChecked] = useState<boolean>(checkedProp || false);
+  const normalizedChecked = checkedProp ?? checked;
 
-    if (checkedProp == null) {
-      setChecked(e.target.checked);
+  const onClick = (e: React.MouseEvent<HTMLButtonElement>) => {
+    if (onClickProp) {
+      onClickProp(e);
     }
+
+    setChecked(curr => {
+      const updatedState = !curr;
+
+      if (onChangeProp) {
```

Comment:
Can do the same here: `onChangeProp?.(updatedState, e)`

Reply:
👍